test/parallel/test-buffer.js

```diff
@@ -1466,13 +1465,22 @@ assert.equal(Buffer.prototype.offset, undefined);
 assert.equal(SlowBuffer.prototype.parent, undefined);
 assert.equal(SlowBuffer.prototype.offset, undefined);
 
-{
-  // Test that large negative Buffer length inputs don't affect the pool offset.
-  assert.deepStrictEqual(Buffer(-Buffer.poolSize), Buffer.from(''));
-  assert.deepStrictEqual(Buffer(-100), Buffer.from(''));
-  assert.deepStrictEqual(Buffer.allocUnsafe(-Buffer.poolSize), Buffer.from(''));
-  assert.deepStrictEqual(Buffer.allocUnsafe(-100), Buffer.from(''));
-
-  // Check pool offset after that by trying to write string into the pool.
-  assert.doesNotThrow(() => Buffer.from('abc'));
-}
```

Comment:
@yorkie The thing is that `allocate()` isn’t really exposed right now as far as I can tell, and #6893 changes lines that right now call `allocUnsafe()` to calling `allocate()`… but I’m not sure it should, so I’d like to have that sorted out before anything here.

And in general, this PR won’t be in a release until v7.0.0, so it doesn’t have to land anytime soon; @RReverser has rebased his PR a couple of times already and there’s no need to lay the burden of resolving another merge conflict upon him. :)